src/call.go

```diff
@@ -0,0 +1,447 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"time"
+
+	"maunium.net/go/mautrix"
+	"maunium.net/go/mautrix/event"
+	"maunium.net/go/mautrix/id"
+
+	"github.com/pion/webrtc/v3"
+)
+
+type Call struct {
+	CallID                 string
+	UserID                 id.UserID
+	DeviceID               id.DeviceID
+	LocalSessionID         id.SessionID
+	RemoteSessionID        id.SessionID
+	Client                 *mautrix.Client
+	PeerConnection         *webrtc.PeerConnection
+	Conf                   *Conference
+	DataChannel            *webrtc.DataChannel
+	LastKeepAliveTimestamp time.Time
```

Comment:
Do all of these need to be public? eg. LastKeepAliveTimestamp doesn't look like its used outside of the class.